test/operator/discr_ops_test.py

```diff
@@ -0,0 +1,242 @@
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External module imports
+import numpy as np
+import pytest
+
+# ODL imports
+from odl.discr.lp_discr import uniform_discr, FunctionSpace, IntervalProd
+from odl.operator.discr_ops import (finite_diff, DiscretePartDeriv,
+                                    DiscreteGradient, DiscreteDivergence)
+from odl.space.ntuples import Rn
+from odl.set.domain import Rectangle
+
+
+def test_finite_diff():
+    """Finite differences test. """
+
+    ndim = 10
+    f = np.arange(ndim, dtype=float)
+
+    df = finite_diff(f)
+    assert all(df == np.ones(ndim))
+
+    dx = 0.5
+    df = finite_diff(f, dx=dx)
+    assert all(df == np.ones(ndim) / dx)
+
+    # central differences with zero padding at boundaries
+    df = finite_diff(f, zero_padding=True)
+    df0 = np.ones(ndim)
+    df0[0] = f[1] / 2.
+    df0[-1] = -f[-2] / 2
+    assert all(df == df0)
+
+    # one-sided difference at boundaries without zero padding
+    df = finite_diff(f, zero_padding=False, edge_order=1)
+    df0 = np.ones(ndim)
+    df0[0] = f[1] - f[0]
+    df0[-1] = f[-1] - f[-2]
+    assert all(df == df0)
+
+    # edge order
+    df1 = finite_diff(np.sin(f / 10 * np.pi), edge_order=1)
+    df2 = finite_diff(np.sin(f / 10 * np.pi), edge_order=2)
+    assert all(df1[1:-1] == df2[1:-1])
+    assert df1[0] is not df2[0]
+    assert df1[-1] is not df2[-1]
+    with pytest.raises(ValueError):
+        finite_diff(f, edge_order=3)
+
+    # axis parameter
+    ndim = 5
+    f = np.arange(ndim, dtype=float)
+    f = f * f.reshape((ndim, 1))
+    df = finite_diff(f, axis=0)
+    f0 = np.array([[-0., 1., 2., 3., 4.],
+                   [0., 1., 2., 3., 4.],
+                   [0., 1., 2., 3., 4.],
+                   [0., 1., 2., 3., 4.],
+                   [0., 1., 2., 3., 4.]])
+    assert (df == f0).all()
+    df = finite_diff(f, axis=1)
+    assert (df == f0.T).all()
+    with pytest.raises(IndexError):
+        finite_diff(f, axis=2)
+
+    # zero padding uses second-order accurate edges
+    with pytest.raises(ValueError):
+        finite_diff(f, zero_padding=True, edge_order=1)
+
+    # at lease two elements are required
+    with pytest.raises(ValueError):
+        finite_diff(np.array([0.0]))
+
+    # in-place evaluation
+    ndim = 10
+    f = np.arange(ndim, dtype=float)
+    out = np.zeros_like(f)
+    assert out is not finite_diff(f)
```

Comment:
There is no way this assert could fail, `finite_diff` doesn't know about `out`, so Id say this is redundant.